### PR TITLE
models: Celery status response from event API

### LINKED ISSUE #40: models: improper status response
Regarding the status metadata propagation from the Event REST API:(https://github.com/inveniosoftware/invenio-webhooks/blob/master/invenio_webhooks/models.py#L176-L181)

The state metadata from the Celery backend should not be propagated only when the state is `PENDING`, because:

1. If `CELERY_TRACK_STARTED` is set, then the state will always be `STARTED ` whenever a meaningful status update occurs.
2. We should allow custom states to expose their status metadata.

Moreover, I think we should not expose only `meta['message']`, but the whole dictionary (e.g. when you do not have access to the code that updates the status).

### PR DESCRIPTION
**Pull request by omelkonian**
* Fixes improper Celery status response through the Event API.
  (closes #40)

Signed-off-by: Orestis Melkonian <melkon.or@gmail.com>

### CODE REVIEW

File: invenio_webhooks/models.py [hunk 1]
```diff
@@ -174,10 +174,8 @@ def status(self, event):
         """Return a tuple with current processing status code and message."""
         result = AsyncResult(str(event.id))
         return (
-            self.CELERY_STATES_TO_HTTP.get(result.state),
-            result.info.get('message')
-            if result.state == states.PENDING and result.info
```
Comment (jirikuncar) [inline invenio_webhooks/models.py]: We could add `states.STARTED` to the condition. But we should use the `event.response` anytime the task is finished or failed.

Comment (jirikuncar) [inline invenio_webhooks/models.py]: When the task is finished the state will be committed in the `event.response`.

Comment (omelkonian) [inline invenio_webhooks/models.py]: Is this currently being done automatically? Or should I implement it?

**Comment by omelkonian** [comment]
> When the task is finished the state will be committed in the event.response.

Is this being done currently?

**Comment by jirikuncar** [comment]
>> When the task is finished the state will be committed in the event.response.
> Is this being done currently?

Yes, if one does `event.response['message'] = 'foo'` somewhere in `run`.

**Comment by omelkonian** [comment]
We could also do it automatically, e.g. by using handlers (http://docs.celeryproject.org/en/latest/userguide/signals.html#std:signal-task_postrun).

File: invenio_webhooks/models.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def status(self, event):
         return (
             self.CELERY_STATES_TO_HTTP.get(result.state),
             result.info.get('message')
-            if result.state == states.PENDING and result.info
+            if result.state in [states.PENDING, states.STARTED] and result.info
```
Comment (jirikuncar) [inline invenio_webhooks/models.py]: `in {states.PENDING, states.STARTED}` for constant lookup times

plus you can move it as class property `CELERY_RESULT_INFO_FOR = {states.PENDING, states.STARTED}` so if you really want to change the behavior for specific receiver you can override it easily.

**Comment by jirikuncar** [comment]
LGTM

**Pull request merged**